manifests/params.pp

```diff
@@ -24,34 +24,62 @@
   # parameters dealing with installation
   $install_server = true
   $install_client = true
-  $version = 'LATEST'
-
+  $version = $::operatingsystem ? {
+    'Ubuntu' => '3.6',
+    default  => 'LATEST',
+  }
   # by default, we'll use the upstream repository
-  $repo    = true
-  $repo_gpg_key_name = 'RPM-GPG-KEY-gluster.pub'
-  $repo_gpg_key_path = '/etc/pki/rpm-gpg/'
-  $repo_gpg_key_source = "puppet:///modules/${module_name}/${repo_gpg_key_name}"
+  $repo = true
   # we explicitly do NOT set a priority here. The user must define
   # a priority in order to ensure that it is activated
   $repo_priority = undef
 
-  # these packages are the upstream names
-  $server_package = 'glusterfs-server'
-  $client_package = 'glusterfs-fuse'
+  # Set distro/release specific names, etc.
+  case $::osfamily {
+    'RedHat': {
+      $repo_gpg_key_name = 'RPM-GPG-KEY-gluster.pub'
+      $repo_gpg_key_path = '/etc/pki/rpm-gpg/'
+      $repo_gpg_key_source = "puppet:///modules/${module_name}/${repo_gpg_key_name}"
 
-  # and these packages are vendor-defined names
-  if $::osfamily == 'RedHat' {
-    $vendor_server_package = $::operatingsystemmajrelease ? {
-      # RHEL 6 and 7 provide Gluster packages natively
-      /(6|7)/ => 'glusterfs',
-      default => false
+      # these packages are the upstream names
+      $server_package = 'glusterfs-server'
+      $client_package = 'glusterfs-fuse'
+      
+      $vendor_server_package = $::operatingsystemmajrelease ? {
+        # RHEL 6 and 7 provide Gluster packages natively
+        /(6|7)/ => 'glusterfs',
+        default => false
+      }
+      $vendor_client_package = $::operatingsystemmajrelease ? {
+        /(6|7)/ => 'glusterfs-fuse',
+        default => false,
+      }
+      
+      $service_name = 'glusterd'
     }
-    $vendor_client_package = $::operatingsystemmajrelease ? {
-      /(6|7)/ => 'glusterfs-fuse',
-      default => false,
+    'Debian': {
+      case $::operatingsystem {
+        'Debian': {
+          # warning( "Need to check this Debian repo key fingerprint.")
+          $repo_gpg_key_name = 'A4703C37D3F4DE7F1819E980FE79BB52D5DC52DC'
+        }
+        default: {
```

Comment:
I'd rather see the `default` value be `Debian`, and an explicit check for `Ubuntu`, since Ubuntu is not Debian.